### PR TITLE
Fix update_flight_costs rental backfill skip logic

### LINKED ISSUE #749: Fix update_flight_costs skipping rental backfill when tow actual is present
Summary\nFinalized flights can retain missing or zero rental_cost_actual values even when rental is computable, because update_flight_costs currently updates only when both tow_cost_actual and rental_cost_actual are missing/zero.\n\nImpact\n- Flights with tow_cost_actual already populated are skipped, leaving rental_cost_actual unset/zero.\n- This caused missing rental charges on tenant-ssc logsheet 2095 (2026-03-07).\n\nRoot cause\nThe command condition uses a combined check requiring both fields to be missing/zero before any write.\n\nProposed fix\n- Update each field independently:\n  - Backfill tow_cost_actual when tow actual is missing/zero.\n  - Backfill rental_cost_actual when rental actual is missing/zero.\n- Add regression tests for:\n  - rental backfill when tow actual is already set\n  - strict --after boundary behavior\n\nAcceptance criteria\n- Running update_flight_costs backfills rental actual even if tow actual is already non-zero.\n- Existing non-zero tow/rental values remain unchanged unless that specific field is missing/zero.\n- New tests pass.

### PR DESCRIPTION
**Pull request by pietbarber**
Closes #749

## Summary
Fix `update_flight_costs` so tow and rental actual costs are backfilled independently.

## Problem
The previous command only updated flights when **both** `tow_cost_actual` and `rental_cost_actual` were missing/zero.

That skipped valid backfills where tow was already set but rental was missing/zero.

## Changes
- Update backfill logic to evaluate each field independently for finalized logsheets only.
- Preserve existing non-zero values unless that specific field is missing/zero.
- Avoid coercing non-computable costs (`None`) to `0.00`.
- Avoid unnecessary writes/count inflation when computed value already matches stored value.
- Add regression tests for:
  - rental backfill when tow actual is already populated
  - strict `--after` boundary behavior (`log_date__gt`)
  - non-computable costs remain `None`
  - unchanged `0.00` rows are not counted as updates

## Validation
- `pytest logsheet/tests/test_update_flight_costs_command.py -q`
- Result: `5 passed`

## Production Context
Observed on `tenant-ssc` logsheet `2095` (`2026-03-07`), where some flights had `tow_cost_actual` set and `rental_cost_actual` missing.

### CODE REVIEW

File: docs/resolved-issues/issue-749-flight-cost-backfill-skip.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+- No schema changes or migrations are required.
+
+## Prevention
+1. Treat independent stored values as independent update targets.
+2. Add regression tests whenever backfill commands combine multiple fields in one guard condition.
+3. Prefer explicit booleans (`should_update_tow`, `should_update_rental`) over compound conditions for clarity.
+4. For production remediations, use dry-run validation queries before applying updates.
+5. Keep `*_actual` fields nullable to preserve unknown vs zero semantics:
+   - `None` means unknown/not locked
+   - `0.00` means known and intentionally zero
```
[the remark this answers is not in the document]
Comment (pietbarber) [reply docs/resolved-issues/issue-749-flight-cost-backfill-skip.md]: Agreed and fixed. I clarified the semantics in the Prevention section to avoid conflict with command behavior:
- `None` means unknown/not locked.
- Historical rows may use `0.00` as a placeholder for missing, and `update_flight_costs` treats `*_cost_actual == 0` like `NULL` for backfill eligibility.

Updated in `docs/resolved-issues/issue-749-flight-cost-backfill-skip.md` (commit `39b0b3c`).

File: docs/resolved-issues/issue-749-flight-cost-backfill-skip.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+- Backfill `rental_cost_actual` when rental actual is missing/zero.
+
+This preserves valid existing data while filling missing fields.
+
+### Tests added
+File: `logsheet/tests/test_update_flight_costs_command.py`
+
+- `test_updates_rental_when_tow_actual_already_set`
+- `test_after_filter_is_strictly_greater_than`
+- `test_does_not_coerce_non_computable_costs_to_zero`
```
[the remark this answers is not in the document]
Comment (pietbarber) [reply docs/resolved-issues/issue-749-flight-cost-backfill-skip.md]: Updated in commit `00f3fe2`.

I added the missing regression test to the resolved-issue doc test list:
- `test_does_not_count_unchanged_zero_costs_as_updates`

File: `docs/resolved-issues/issue-749-flight-cost-backfill-skip.md`.

File: logsheet/management/commands/update_flight_costs.py [hunk 3]
```diff
@@ -29,27 +31,53 @@ def handle(self, *args, **options):
         if not logsheets.exists():
             raise CommandError(f"No logsheets found after {after_str}.")
```
[the remark this answers is not in the document]
Comment (pietbarber) [reply logsheet/management/commands/update_flight_costs.py]: Implemented in commit `d2bd3c1`.

I restricted the backfill command to finalized logsheets only, so draft logsheets are now skipped and cannot have `*_cost_actual` locked prematurely.

Changes:
- `logsheet/management/commands/update_flight_costs.py`: `Logsheet` queryset now filters with `finalized=True` alongside `log_date__gt`.
- `logsheet/tests/test_update_flight_costs_command.py`: added `test_skips_non_finalized_logsheets_when_backfilling` to verify draft flights remain unchanged while finalized flights are backfilled.

Validation:
- `pytest logsheet/tests/test_update_flight_costs_command.py -q` -> `5 passed`.
